rationale.md

```diff
@@ -0,0 +1,61 @@
+# OpenTelemetry Rationale
+
+When creating a library, often times designs and decisions are made that get lost over time. This
+document tries to collect information on design decisions to answer common questions that may come
+up when you explore the SDK.
+
+## Versioning and Releasing
+
+The OpenTelemetry Applications and OpenTelemetry Spec itself use semver v2.
+
+## Goals
+
+### API Stability
+
+Once the API for a given signal (spans, logs, metrics, baggage) has been officially released, that API module will function with any SDK that has the same major version, and equal or greater minor or patch version.
+
+For example, libraries that are instrumented with `opentelemetry-api 1.0.1` will function with SDK library `opentelemetry-sdk 1.11.33` or `opentelemetry-sdk 1.3.4`.
+
+### SDK Stability:
+
+Public portions of the SDK (constructors, configuration, end-user interfaces) must remain backwards compatible.
+Internal interfaces are allowed to break.
```

Comment:
Must they always remain backward compatible? Could an SDK release a v2 or v3 that introduce breaking changes to SDK but still confirms to API 1.x? I know this shouldn't ideally happen but does the spec allow it?

Reply:
This is taken right from the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/71cc619739be46aa0a9f42210f60246bb8b1885a/specification/versioning-and-stability.md#sdk-stability). Spec enforces backwards compatibility between SDK versions. My understanding is that a breaking change requires a major version bump (from 1.X.Y -> 2.0.0).